storage/database.go

```diff
@@ -138,21 +138,15 @@ func NewDatabase(
 	scope := iopts.MetricsScope().SubScope("database")
 	logger := iopts.Logger()
 
-	index := databaseIndexNoOp
-	if opts.IndexingEnabled() {
-		index, err = newDatabaseIndex(opts)
-		if err != nil {
-			return nil, err
-		}
-	}
+	var idx databaseIndex
```

Comment:
Hm this is just nil yeah?  Is it worth explicitly setting just the default nil value on the `db` struct?

Reply:
Sure, moved to the namespace and made the change.